native_lib/DDGSyncCrypto.c

```diff
@@ -0,0 +1,150 @@
+
+#include <string.h>
+
+#include "DDGSyncCrypto.h"
+#include "sodium.h"
+
+// Seems mad that you still need to define this?!
+#define min(x, y) (x < y) ? x : y
+
+// Contexts must be 8 characters long but are otherwise arbitrary
+#define DDGSYNC_STRETCHED_PRIMARY_KEY_CONTEXT "Stretchy"
```

Comment:
Forgive my question, I'm not familiar with libsodium, but, are these hash salts? If they are, is there any security concern about having the raw string values embedded in the binary?

Reply:
I'm not sure what the best practice is tbh. But, if this was externalised then each consumer would have to pass in the same context as a parameter.   All consumers will be open source (eventually) so I don't see any particularly added value in externalising.  I will ping Stefan to validate.

Reply:
These are not hash salts but the "context" which is a string describing what the key is going to be used for. It doesn't have to be secret, just unique for each key derived from the primary key.

Reply:
Thanks for the clarification @stefanorri !